src/components/OptionsSelector.js

```diff
@@ -123,92 +221,93 @@ class OptionsSelector extends Component {
         }
     }
 
-    /**
-     * Scrolls to the focused index within the SectionList
-     *
-     * @param {Number} sectionIndex
-     * @param {Number} itemIndex
-     */
-    scrollToFocusedIndex(sectionIndex, itemIndex) {
-        this.list.scrollToLocation({sectionIndex, itemIndex});
-    }
-
-    /**
-     * Delegate key presses to specific callbacks
-     *
-     * @param {SyntheticEvent} e
-     */
-    handleKeyPress(e) {
-        if (!this.list) {
+    componentDidUpdate(prevProps, prevState) {
+        if (_.isEqual(this.props.sections, prevProps.sections)) {
             return;
         }
 
-        // We are mapping over all the options to combine them into a single array and also saving the section index
-        // index within that section so we can navigate
-        const allOptions = _.reduce(this.props.sections, (options, section, sectionIndex) => (
-            [...options, ..._.map(section.data, (option, index) => ({
-                ...option,
-                index,
-                sectionIndex,
-            }))]
-        ), []);
-
-        if (allOptions.length === 0) {
-            return;
-        }
+        const newOptions = this.flattenSections();
+        const newFocusedIndex = newOptions.length === lodashGet(prevState, 'allOptions.length') ? prevState.focusedIndex : this.props.selectedOptions.length;
```

Comment:
@roryabraham I don't understand this change. It's causing two bugs listed below 🐛🐛 -  

Bug 1: index out of bounds when selecting the only search result.
1. `⌘ + Shift + K`
2. Search `rushatgabhane@gmail.com`
3. Select the user

https://user-images.githubusercontent.com/29673073/171840399-7bf7e5be-b98c-4429-872f-2a0e8753159e.mov

Reply:
Bug 2: selecting a user with mouse doesn't change the focused index to the best result.
1. `⌘ + Shift + K`
2. Select the first user USING MOUSE.
Expected result: The focused index changes to the second user.
Actual result: The focused index stays on the first user.

https://user-images.githubusercontent.com/29673073/171841401-9d84f1ca-6847-4af3-9215-fab68e5dee74.mov

Reply:
Should we revert back to `const newFocusedIndex = this.props.selectedOptions.length`?